lib/bpf/bpf.go

```diff
@@ -364,27 +363,31 @@ func (s *Service) emitCommandEvent(eventBytes []byte) {
 	// Args are sent in their own event by execsnoop to save stack space. Store
 	// the args in a ttlmap, so they can be retrieved when the return event arrives.
 	case eventArg:
-		var buf []string
-		buffer, ok := s.argsCache.Get(strconv.FormatUint(event.PID, 10))
-		if !ok {
-			buf = make([]string, 0)
-		} else {
-			buf = buffer.([]string)
-		}
+		key := strconv.FormatUint(event.PID, 10)
+
+		// loadFn does not return an error, so we can skip error handling.
+		args, _ := utils.FnCacheGet(s.closeContext, s.argsCache, key, func(ctx context.Context) ([]string, error) {
+			return make([]string, 0), nil
+		})
```

Comment:
nit: it's very easy to accidentally break this invariant elsewhere. Consider at least having a `this is a bug` log get triggered in the event of a non-nil error.